hyperion/app/creation_steps/workspace/git_manager.py

```
@@ -0,0 +1,725 @@
+            result = self._run_git_command(context, ["init"])
+            if result.returncode != 0:
+                raise GitException(f"Git init failed: {result.stderr}")
+
+            # Set up initial configuration
+            self._setup_initial_config(context)
+
+            logger.info("Git repository initialized successfully")
+
+        except Exception as e:
+            logger.error(f"Failed to initialize Git repository: {e}")
+            raise GitException(f"Failed to initialize Git repository: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)